roles/foreman_provisioning/tasks/main.yml

```diff
@@ -145,82 +144,78 @@
+- name: 'Setup CentOS 7 provisioning'
+  import_tasks: configure_centos_7.yml
 
-- name: 'set kickstart templates'
-  set_fact:
-    kickstart_templates: "{{ kickstart_templates_json.stdout|from_json }}"
+- name: 'Setup Fedora 27 provisioning'
+  import_tasks: configure_fedora_27.yml
 
-- name: 'associate kickstart templates to CentOS 7'
-  shell: >
-    {{ foreman_provisioning_hammer }} template add-operatingsystem --id {{ item.Id }} --operatingsystem 'CentOS 7'
-  with_items: "{{ kickstart_templates }}"
+- name: 'Setup Debian 9.3 provisioning'
+  import_tasks: configure_debian_9_3.yml
 
-- name: 'set default templates for CentOS 7'
-  shell: >
-    {{ foreman_provisioning_hammer }} os set-default-template --id {{ foreman_provisioning_centos73.Id }} --config-template-id {{ item.Id }}
-  with_items: "{{ kickstart_templates }}"
+- name: 'Setup Ubuntu 17.10 provisioning'
+  import_tasks: configure_ubuntu_17_10.yml
 
+# TODO verify - should not be needed, DNS should work just fine
 - name: 'set unattended_url'
   shell: >
     {{ foreman_provisioning_hammer }} settings set --name unattended_url --value 'http://{{ foreman_provisioning_ip_address }}'
 
-- name: 'find hostgroup Forklift Base'
+- name: 'find hostgroup Base'
   shell: >
-    {{ foreman_provisioning_hammer }} hostgroup info --name 'Forklift Base'
+    {{ foreman_provisioning_hammer }} hostgroup info --name 'Base'
   register: foreman_provisioning_hostgroup_base
   ignore_errors: True
 
-- name: 'create hostgroup Forklift Base'
+# TODO can't set environment http://projects.theforeman.org/issues/22517
+# when it's fixed add this:
+# --environment production
+# TODO compute-profile can't be specified by name until http://projects.theforeman.org/issues/21580/ so we hardcode 1
+- name: 'create hostgroup Base'
   shell: >
     {{ foreman_provisioning_hammer }} hostgroup create
-    --name 'Forklift Base'
+    --name 'Base'
     --architecture x86_64
     --domain {{ foreman_provisioning_domain }}
-    --environment production
     --puppet-ca-proxy-id {{ foreman_provisioning_smart_proxy.Id }}
     --puppet-proxy-id {{ foreman_provisioning_smart_proxy.Id }}
     --subnet '{{ foreman_provisioning_network }}/24'
+    --compute-profile-id 1
+    --compute-resource 'libvirt'
     --root-pass changeme
     {{ foreman_provisioning_hammer_taxonomy_params }}
   when: foreman_provisioning_hostgroup_base.stderr.find('not found') != -1
 
-- name: 'find hostgroup Forklift CentOS 7'
+- name: 'find hostgroup CentOS 7'
   shell: >
-    {{ foreman_provisioning_hammer }} hostgroup info --name 'Forklift CentOS 7 Mirror'
+    {{ foreman_provisioning_hammer }} hostgroup info --name 'CentOS 7 Mirror'
   register: foreman_provisioning_hostgroup_centos_mirror
   ignore_errors: True
 
-- name: 'create hostgroup Forklift CentOS 7 Mirror'
+- name: 'create hostgroup CentOS 7 Mirror'
   shell: >
     {{ foreman_provisioning_hammer }} hostgroup create
-    --name 'Forklift CentOS 7 Mirror'
+    --name 'CentOS 7 Mirror'
```

Comment:
I guess we should add `--pxe-loader "PXELinux BIOS"` here, as otherwise, the provisioning doesn't work out of the box

Reply:
added to the base rather, it gets inherited to all children

Reply:
:+1: interesting it was not possible to set on UI, when OS was not selected, therefore I was not suggesting that. Anyway, tested by updating the loader just on the base and it was properly propagating.